org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/CodeTemplatePreferences.java

```diff
@@ -191,4 +196,47 @@
 	 */
 	public static final String CODETEMPLATE_RECORDSNIPPET_PUBLIC = "${package_header}/**\n * ${type_name}\n */\npublic record ${type_name}(${cursor}) {\n}";
 
+	public static class Month extends SimpleTemplateVariableResolver {
+		public Month() {
+			super("month", "Current month"); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+
+		@Override
+		protected String resolve(TemplateContext context) {
+			return Integer.toString(Calendar.getInstance().get(Calendar.MONTH));
```

Comment:
The generated month value is from 0-11, we need to change it to 1-starting.

Reply:
and both month and day should use 2 digits: 2021-08-01

Reply:
Yup. Looks like I did it in https://github.com/redhat-developer/vscode-java/issues/2052#issuecomment-897985236, but forgot to do the same when I copied it over. I'll prefix single digit months with 0 since I like that approach also.